### PR TITLE
(fix) add further instructions for downloading in Storage.get docs

### PR DESCRIPTION
**Pull request by sw-yx**
*Description of changes:* (fix) add further instructions for downloading in Storage.get docs

closes https://github.com/aws-amplify/docs/issues/2512

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/lib/storage/fragments/js/download.md [hunk 1]
```diff
@@ -23,12 +42,37 @@ Send object data for immediate file download:
...
+// data.Body is a Blob
 data.Body.text().then(string => { 
   // handle the String data return String 
 })
 ```
 
+You can programmatically download Blobs using JavaScript:
+
+```js
+// https://blog.logrocket.com/programmatic-file-downloads-in-the-browser-9a5186298d5c/
```
Comment (sammartinez) [inline docs/lib/storage/fragments/js/download.md:54]: Is this ok to use ? Does the person that owns the blog understand we would use this? I don't want to recommend a blog link if it leads to the person will remove it. Can we look to have this remove?

Comment (sw-yx) [reply docs/lib/storage/fragments/js/download.md]: done!

File: docs/lib/storage/fragments/js/download.md [hunk 2]
```diff
@@ -23,12 +42,36 @@ Send object data for immediate file download:
...
+    }, 150);
+  };
+  a.addEventListener('click', clickHandler, false);
+  a.click();
+  return a;
+}
+
+// usage
+Storage.get(fileKey, { download: true })
+        .then(res => downloadBlob(res.Body, downloadFileName)) // can derive from fileKey if you wish
```
Comment (sw-yx) [inline docs/lib/storage/fragments/js/download.md:72]:
```suggestion
        .then(res => downloadBlob(res.Body, downloadFileName)) // derive downloadFileName from fileKey if you wish
```

File: docs/lib/storage/fragments/js/download.md [hunk 3]
```diff
@@ -14,7 +14,26 @@ await Storage.get(key: string, config: {
 
 `Storage.get` returns a signed URL `string` to your file if `download` is false, which is the default.
 
-`Storage.get` returns a [`Blob`](https://developer.mozilla.org/en-US/docs/Web/API/Blob) of object data if `download` is true.
+`Storage.get` returns an object with a `Body` of [`Blob`](https://developer.mozilla.org/en-US/docs/Web/API/Blob) of object data if `download` is true. The full return signature looks like this:
```
Comment (sw-yx) [inline docs/lib/storage/fragments/js/download.md:17]:
```suggestion
If `download` is true, `Storage.get` returns an object with a `Body` field of type [`Blob`](https://developer.mozilla.org/en-US/docs/Web/API/Blob). It is up to you to then save it to disk (we suggest a method below). 

The full return signature looks like this:
```

File: docs/lib/storage/fragments/js/download.md [hunk 4]
```diff
@@ -12,9 +12,30 @@ await Storage.get(key: string, config: {
 })
 ```
 
-`Storage.get` returns a signed URL `string` to your file if `download` is false, which is the default.
-
-`Storage.get` returns a [`Blob`](https://developer.mozilla.org/en-US/docs/Web/API/Blob) of object data if `download` is true.
+`Storage.get` merely returns a signed URL `string` to your file, if `download` is false, which is the default. You can use this to create a download link for people to click. This is usually the recommended option. Note that this method **does not check if the file actually exists** as that would involve an extra API call.
```
Comment (sammartinez) [inline docs/lib/storage/fragments/js/download.md:15]:
```suggestion
`Storage.get` returns a signed URL `string` to your file, if `download` is false, which is the default. You can use this to create a download link for people to click on. This is our recommended option. Note that this method **does not check if the file actually exists** as that would involve an extra API call.
```

**Review by sammartinez — APPROVED** [review]
LGTM 🌮  Thanks for getting the updates done @sw-yx !

File: docs/lib/storage/fragments/js/download.md [hunk 5]
```diff
@@ -12,25 +12,78 @@ await Storage.get(key: string, config: {
...
-Send object data for immediate file download:
+### File download option
+
+The `download` option send you the object data for download or programmatic manipulation:
 
 ```javascript
 const data = await Storage.get(`filename.txt`, { download: true })
 
-// data is a Blob
+// data.Body is a Blob
```
Comment (sw-yx) [inline docs/lib/storage/fragments/js/download.md:34]: this was a mistake by me in my first PR to this doc, which led to https://github.com/aws-amplify/docs/issues/2512

File: docs/lib/storage/fragments/js/download.md [hunk 6]
```diff
@@ -12,25 +12,78 @@ await Storage.get(key: string, config: {
 })
 ```
 
-`Storage.get` returns a signed URL `string` to your file if `download` is false, which is the default.
+`Storage.get` returns a signed URL `string` to your file, if `download` is false, which is the default. You can use this to create a download link for people to click on. This is our recommended option. Note that this method **does not check if the file actually exists** as that would involve an extra API call.
 
-`Storage.get` returns a [`Blob`](https://developer.mozilla.org/en-US/docs/Web/API/Blob) of object data if `download` is true.
+```js
+// get the signed URL string
+const signedURL = await Storage.get(file.__data.key)
```
Comment (sw-yx) [inline docs/lib/storage/fragments/js/download.md:19]:
```suggestion
const signedURL = await Storage.get(key) // get key from Storage.list
```

File: docs/lib/storage/fragments/js/download.md [hunk 7]
```diff
@@ -12,25 +12,78 @@ await Storage.get(key: string, config: {
...
+  a.click();
+  return a;
+}
+
+// usage
+Storage.get(fileKey, { download: true })
+        .then(res => downloadBlob(res.Body, downloadFileName)) // derive downloadFileName from fileKey if you wish
+```
+
+The full return signature of `Storage.get(somekey, { download: true })` looks like this:
```
Comment (sw-yx) [inline docs/lib/storage/fragments/js/download.md:66]:
```suggestion
The full return signature of `Storage.get(key, { download: true })` looks like this:
```

**Pull request merged**